runtime/runtime.go

```diff
@@ -1163,7 +1163,14 @@ func (r *interpreterRuntime) instantiateContract(
 	parameterCount := len(parameterTypes)
 
 	if argumentCount != parameterCount {
-		return nil, fmt.Errorf("invalid argument count: expected %d, got %d", parameterCount, argumentCount)
+		if argumentCount < parameterCount {
+			return nil, fmt.Errorf("invalid argument count, too few arguments: expected %d, got %d, next missing argument: `%s`",
+				parameterCount, argumentCount,
+				parameterTypes[argumentCount])
+		} else if argumentCount > parameterCount {
+			return nil, fmt.Errorf("invalid argument count, too many arguments: expected %d, got %d",
+				parameterCount, argumentCount)
```

Comment:
Chopping down the arguments could make this a bit more readable:

```suggestion
			return nil, fmt.Errorf(
				"invalid argument count, too few arguments: expected %d, got %d, next missing argument: `%s`",
				parameterCount, 
				argumentCount,
				parameterTypes[argumentCount],
			)
		} else if argumentCount > parameterCount {
			return nil, fmt.Errorf(
				"invalid argument count, too many arguments: expected %d, got %d",
				parameterCount, 
				argumentCount,
			)
```